alembic/versions/20240821_7a2fcaac8b63_add_loan_identifier_column_to_playtime_tables.py

```diff
@@ -22,12 +22,29 @@ def upgrade() -> None:
 
     op.add_column(
         "playtime_entries",
-        sa.Column("loan_identifier", sa.String(length=40), nullable=False, default=""),
+        sa.Column("loan_identifier", sa.String(length=40), nullable=True, default=""),
     )
 
     op.add_column(
         "playtime_summaries",
-        sa.Column("loan_identifier", sa.String(length=40), nullable=False, default=""),
+        sa.Column("loan_identifier", sa.String(length=40), nullable=True, default=""),
```

Comment:
This looks good and will definitely do the job.

I do think we could resolve the issue without the explicit `UPDATE`s or additional `ALTER COLUMN`s by setting the server_default, as follows:
```python
    op.add_column(
        "playtime_entries",
        sa.Column(
            "loan_identifier", 
            sa.String(length=40), 
            nullable=False,  # <-
            default="",
            server_default="",  # <-
        ),
    )
    op.add_column(
        "playtime_summaries",
        sa.Column(
            "loan_identifier", 
            sa.String(length=40), 
            nullable=False,  # <-
            default="",
            server_default="",  # <-
        ),
    )
```

The `server_default` should affect only the existing rows.

Reply:
This suggestion would work as well. 

In order to not have alembic show a DB difference, we would have to alter the column after to remove the `server_default` or update the sqlalchemy model to also contain a `server_default`. Since the it gets set on the column in the DB, so adding a default is actually a change to the schema.